packages/langium/src/parser/lexer.ts

```diff
@@ -25,9 +25,17 @@ export interface LexerResult {
     report?: LexingReport;
 }
 
+export type TokenizeMode = 'full' | 'partial';
+
+export interface TokenizeOptions {
+    mode: TokenizeMode;
```

Comment:
Suggestion: With these kinds of option interfaces (especially if they're optional), their properties should be optional as well.
```suggestion
    mode?: TokenizeMode;
```